django/core/cache/utils.py

```diff
@@ -8,5 +8,13 @@ def make_template_fragment_key(fragment_name, vary_on=None):
     if vary_on is None:
         vary_on = ()
     key = ':'.join(quote(str(var)) for var in vary_on)
-    args = hashlib.md5(key.encode())
+
+    # Try to generate an MD5 hash but fall up to SHA256 if in FIPS mode
+    try:
+        args = hashlib.md5(key.encode())
```

Comment:
Rather than try this each time could we not include the `is_fips_mode` method in this module and do a conditional check? We can wrap it in an `@lru_cache` to ensure it's not called more than once.